critiquebrainz/db/oauth_client.py

```diff
@@ -0,0 +1,137 @@
+        connection.execute(sqlalchemy.text("""
+            INSERT INTO oauth_client VALUES(:client_id, :client_secret, :redirect_uri,
+                :user_id, :name, :desc, :website)
+        """), client)
+
+    return client
+
+
+def update(*, client_id, name=None, desc=None, website=None, redirect_uri=None):
+    """Update information related to a OAuth client
+
+    Args:
+        client_id(str): Client ID.
+        name(str, optional): Updated name of the client.
+        desc(str, optional): Updated description of the client.
+        website(str, optional): Client web site.
+        redirect_uri(str, optional): URI where responses will be sent.
+    """
+    updates = []
+    update_data = {}
+    if name is not None:
+        updates.append("name = :name")
+        update_data["name"] = name
+    if desc is not None:
+        updates.append(""" "desc" = :desc""")
+        update_data["desc"] = desc
+    if website is not None:
+        updates.append("website = :website")
+        update_data["website"] = website
+    if redirect_uri is not None:
+        updates.append("redirect_uri = :redirect_uri")
+        update_data["redirect_uri"] = redirect_uri
+
+    update_str = ", ".join(updates)
+    update_data["client_id"] = client_id
+
+    query = sqlalchemy.text("""
+        UPDATE oauth_client
+           SET {update_str}
+         WHERE client_id = :client_id
+    """.format(update_str=update_str))
+
+    if update_str:
+        with db.engine.connect() as connection:
+            connection.execute(query, update_data)
+
+
+def delete(client_id):
+    """Delete OAuth client.
+
+    Args:
+        client_id(str): ID of the OAuth Client.
+    """
+    with db.engine.connect() as connection:
+        connection.execute(sqlalchemy.text("""
+            DELETE
+              FROM oauth_client
+             WHERE client_id = :client_id
+        """), {
+            "client_id": client_id,
+        })
+
+
+def get_client(client_id):
+    """Get info about an OAuth Client.
+
+    Args:
+        client_id(str): ID of the client.
+    Returns:
+        Dict with the following structure:
+        {
+            "client_id": str,
+            "client_secret": str,
+            "redirect_uri": str,
+            "user_id": uuid,
+            "name": str,
+            "desc": str,
+            "website": str,
+        }
+    """
+    with db.engine.connect() as connection:
+        result = connection.execute(sqlalchemy.text("""
+            SELECT client_id,
+                   client_secret,
+                   redirect_uri,
+                   user_id,
+                   name,
+                   "desc",
+                   website
+              FROM oauth_client
+             WHERE client_id = :client_id
+        """), {
+            "client_id": client_id,
+        })
+        row = result.fetchone()
+        if not row:
+            raise db_exceptions.NoDataFoundException("No client with ID: {} found".format(client_id))
```

Comment:
Minor thing, but I'd expect text after `:` to be just the ID. `Can't find OAuth client with ID: %(id)s` would be a better message, I think.